### PR TITLE
Use `jest-worker`.

### LINKED ISSUE #23: Switch from `worker-farm` to `jest-worker`?
In Jest 22 the team exposed the internal `worker-farm` equivalent, [`jest-worker`](https://github.com/facebook/jest/tree/master/packages/jest-worker). Would it be worth switching from `worker-farm` to `jest-worker` for consistency with Jest's default runner?

### PR DESCRIPTION
**Pull request by wtgtybhertgeghgtwtg**
Maybe closes #23.  I'm not too confident about it, though.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,6 +1,5 @@
 const throat = require('throat');
-const pify = require('pify');
-const workerFarm = require('worker-farm');
+const Worker = require('jest-worker').default;
```
Comment (ljharb) [inline src/index.js]: `.default` is a babel implementation detail, and should never be exposed in an entry point. is there a bug filed on `jest-worker` for this?

File: src/runESLint.js [hunk 2]
```diff
@@ -59,43 +59,32 @@ const pass = ({ start, end, testPath }) =>
     jestTestPath: testPath,
   });
 
-const runESLint = ({ testPath, config }, workerCallback) => {
-  try {
-    const start = +new Date();
+module.exports.runESLint = ({ testPath, config }) => {
+  const start = +new Date();
```
Comment (ljharb) [inline src/runESLint.js]: `Date.now()`

File: src/runESLint.js [hunk 3]
```diff
@@ -59,43 +59,32 @@ const pass = ({ start, end, testPath }) =>
...
+  }
 
-      if (options.cliOptions && options.cliOptions.fix) {
-        CLIEngine.outputFixes(report);
-      }
+  const { CLIEngine } = getLocalESLint(config);
+  const options = getESLintOptions(config);
+  const cli = new CLIEngine(options.cliOptions);
+  if (cli.isPathIgnored(testPath)) {
+    const end = +new Date();
```
Comment (ljharb) [inline src/runESLint.js]: also `Date.now()`

File: src/runESLint.js [hunk 4]
```diff
@@ -59,43 +59,32 @@ const pass = ({ start, end, testPath }) =>
...
+  if (options.cliOptions && options.cliOptions.fix) {
+    CLIEngine.outputFixes(report);
+  }
 
-      if (report.errorCount > 0) {
-        const formatter = cli.getFormatter(options.cliOptions.format);
-        const errorMessage = formatter(
-          CLIEngine.getErrorResults(report.results),
-        );
+  const end = +new Date();
```
Comment (ljharb) [inline src/runESLint.js]: also here

File: src/index.js (hunk 1 above)
Comment (wtgtybhertgeghgtwtg) [reply src/index.js]: I don't see any.

File: src/runESLint.js (hunk 2 above)
Comment (wtgtybhertgeghgtwtg) [reply src/runESLint.js]: It was like that when I got here.  Should I still fix it?

File: src/index.js (hunk 1 above)
Comment (ljharb) [reply src/index.js]: Would you mind filing one and linking to it here?

File: src/runESLint.js (hunk 2 above)
Comment (ljharb) [reply src/runESLint.js]: Ah, fair enough. Not required, but it’d be nice :-)

File: src/index.js (hunk 1 above)
Comment (wtgtybhertgeghgtwtg) [reply src/index.js]: I'm not very good with words, but I don't mind.  What should I say in this issue?

Comment (midzelis) [reply src/index.js]: Whats up with this patch. It seems good to me. I don't see an issue with the .default thing. In fact, its how https://github.com/rogeliog/create-jest-runner/blob/master/lib/createJestRunner.js#L1 does it, which was recommended in issue #23

Comment (ljharb) [reply src/index.js]: I suppose it doesn't have to block this issue; but it's indeed a huge problem for babel interop details to pollute the ecosystem.

I've filed https://github.com/facebook/jest/issues/5803

**Pull request closed without merging**